src/command_modules/azure-cli-redis/azure/cli/command_modules/redis/custom.py

```diff
@@ -108,20 +78,64 @@ def cli_redis_create(cmd, client,
     :param enable_non_ssl_port: If the value is true, then the non-ssl redis server port (6379) will be enabled.
     :param tenant_settings: Json dictionary with tenant settings
     :param shard_count: The number of shards to be created on a Premium Cluster Cache.
+    :param minimum_tls_version: Optional: requires clients to use a specified
+     TLS version (or higher) to connect (e,g, '1.0', '1.1', '1.2'). Possible
+     values include: '1.0', '1.1', '1.2'
     :param subnet_id: The full resource ID of a subnet in a virtual network to deploy the redis cache in. Example format /subscriptions/{subid}/resourceGroups/{resourceGroupName}/Microsoft.{Network|ClassicNetwork}/VirtualNetworks/vnet1/subnets/subnet1
     :param static_ip: Required when deploying a redis cache inside an existing Azure Virtual Network.
+    :param zones: A list of availability zones denoting where the resource needs to come from.
+    :param tags: Json dictionary with Resource tags. Example : {\"testKey\":\"testValue\"}
     """
     from azure.mgmt.redis.models import RedisCreateParameters, Sku
+    # pylint: disable=too-many-function-args
     params = RedisCreateParameters(
-        location=location,
-        sku=Sku(sku, vm_size[0], vm_size[1:]),
-        tags=tags,
+        Sku(sku, vm_size[0], vm_size[1:]),
+        location,
         redis_configuration=redis_configuration,
         enable_non_ssl_port=enable_non_ssl_port,
         tenant_settings=tenant_settings,
         shard_count=shard_count,
+        minimum_tls_version=minimum_tls_version,
         subnet_id=subnet_id,
-        static_ip=static_ip)
+        static_ip=static_ip,
+        zones=zones,
+        tags=tags)
     return client.create(resource_group_name, name, params)
 
+
+# pylint: disable=unused-argument
+def cli_redis_create_server_link(cmd, client, primary_cache, secondary_cache):
+    redis_client = cf_redis(cmd.cli_ctx)
+    primary_cache_resource = get_cache_from_resource_id(redis_client, primary_cache)
```

Comment:
It could be helpful to support using name and not just id for the caches.

See [here](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_command_modules/authoring_commands.md#supporting-name-or-id-parameters).

Reply:
I understand that in case of a single name argument because --resource-group is a  mandatory argument. But in this case, user needs to give details about two different caches. They could be of different resource groups. So to implement name or resource id, I'll need to add a third argument for resource group of secondary cache. If i'm doing that, then I don't need the resource Id. In that case, i will have 4 arguments: resource group of primary cache, name of primary cache, resource group of secondary cache, name of secondary cache. But for the current implementation, user needs to give just two values.

Reply:
To clarify with name or id, you can either accept both ids or if a name is given for any of the arguments. In this case, if name is given for either cache, you assume it is in the resource group that was passed in by the user, as mentioned in the link above. I wasn't suggesting you had more resource group arguments.

So `-g MyResourceGroup --primary-cache {ResourceID} --secondary-cache cache2`
Would assume `cache2` is in `MyResourceGroup` and build the ID from the resource group.